.github/workflows/pytest.yaml

```
@@ -1,51 +1,25 @@
-name: Run pytest
+name: pytest
 
-on: [push]
+on:
+  push:
+    branch: master
+  pull_request:
+  schedule:
+    - cron: '0 2 * * *'
+  
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  build:
-
+  pytest:
     runs-on: ubuntu-latest
     container: graphcore/pytorch:latest
-    outputs:
-      summary: ${{ steps.output.outputs.summary }}
-    env:
-      VIRTUAL_ENV: /some/fake/venv/path
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
-          pip install -r requirements-dev.txt
-          pip install -r requirements-jupyter.txt
-          pip install -r requirements-precommit.txt
-          : #Install virtualenv outside of requirements file to avoid installing on local machines
-          pip install virtualenv
-
-      - name: Test with pytest
-        run: |
-          python -m pytest --cov --cov-report xml --junitxml=reports/pytest_report.xml
-      - name: Upload Report
-        if: always()
-        uses: actions/upload-artifact@v3
-        with:
-          name: reports_${{ github.run_id }}
-          path: reports/*report.xml
-      - name: Upload Report
-        if: always()
-        uses: actions/upload-artifact@v3
-        with:
-          name: coverage_${{ github.run_id }}
-          path: coverage.xml
-      - name: Publish Summary
+      - run: python -m pip install . && pip install -r requirements-dev.txt
```

There's a slim chance that `python -m pip ... & pip ...` will call different `pip`s.

You should just be able to use system python from the container:

```suggestion
      - run: pip install . && pip install -r requirements-dev.txt
```